### PR TITLE
Meta to front matter

### PR DESCRIPTION
**Pull request by wbamberg**
Fixes https://github.com/mdn/stumptown-content/issues/69.

Sorry this is such a mega-PR. I was going to change just video's content and the script, and then change everything else later. But then I realized `related_content` would not work properly without updating everything.

So this change should:
* delete `meta.yaml` for all files under html/reference/elements
* move the contents of `meta.yaml` into `prose.md` as front matter, and renamed the result to `docs.md`
* updated all the build-json code to reflect this change

I have also updated all keys in front matter to use `_` rather than `-`: so, `mdn-url`->`mdn_url` etc. This wasn't needed but while I was in there it seemed like a good point to make this change.

I have verified that this produces identical JSON output to the old code.

I have also written a little test tool that does some basic validation of docs.md against its recipe. I used this to catch errors in this PR, but didn't include the tool itself in the PR. Perhaps it could be a part of https://github.com/mdn/stumptown-content/issues/90?

This work also made me think: should we include `title` and `mdn_url` in the html-element recipe?

### CODE REVIEW

File: content/html/reference/elements/abbr/docs.md [hunk 1]
```diff
@@ -1,3 +1,24 @@
+---
+title: '<abbr>: The Abbreviation element'
+mdn_url: https://developer.mozilla.org/docs/Web/HTML/Element/abbr
+tags:
+    group: Inline text semantics
+api: HTMLElement
+permitted_aria_roles:
+    - any
+tag_omission: none
```
Comment (peterbe) [inline content/html/reference/elements/abbr/docs.md]: What are all these (`tags`, `api`, `permitted_aria_roles`, `tag_omission`)??

By the way, in the new read-only MDN site (e.g. beta.developer.mozilla.org) we have removed all display of tags.

**Comment by peterbe** [comment]
> This work also made me think: should we include `title` and `mdn_url` in the html-element recipe?

Nah. I think you can hardcode those in the validator/linter.

File: content/html/reference/elements/abbr/docs.md (hunk 1 above)
Comment (wbamberg) [reply content/html/reference/elements/abbr/docs.md]: `api`, `permitted_aria_roles`, `tag_omission` are all things that live in the "technical summary" box (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video#Technical_summary).

We didn't think too hard about these when we added them to the recipe, and I'm not sure if we really want all of them to be represented as data (yes on `api` I think, and perhaps `permitted_aria_roles`, but `tag_omission` is tricky to represent structured). Also I think we might want a better name than `api` - it's called "DOM Interface" in the page.

`tags` is just generic tags. We don't have a use for tags yet but I expect we might in future.

> By the way, in the new read-only MDN site (e.g. beta.developer.mozilla.org) we have removed all display of tags.

\o/. Tags in Kuma are a [horrible mess](https://developer.mozilla.org/en-US/docs/tags).

Comment (peterbe) [reply content/html/reference/elements/abbr/docs.md]: I suppose we, like with examples, make that a `tech_summary.yaml` file in the `html/reference/element/video/` directory and have a React "ingredient" component to render it out. But not a blocker for progress.

**Comment by peterbe** [comment]
>  I'd love to see a move to `tagname.md` instead of `prose.md` or `docs.md`

I agree. I hate when you have many folders, in a javascript project, all filled with `index.js` files and your IDE tabs become less useful. But if it's a hassle don't let it block.

**Comment by wbamberg** [comment]
> This seems fine to me, though I'd love to see a move to `tagname.md` instead of `prose.md` or `docs.md`

This is not as simple as it seems. Currently the build-json and linting code code [rely on specific filenames to find the things they need to build/lint](https://github.com/mdn/stumptown-content/blob/master/scripts/build-json/build-json.js#L10). We can't just process every .md file under /content because there are also "supplementary files" like attribute descriptions and example descriptions.

So I guess (mostly unpalatable) options include:

* continue to use magic names for files. This has the drawback you pointed out and it also fragile and implicit: if someone happens to call a different file docs.md then things will break.

* only allow these special kinds of .md files under /content, and move other .md files somewhere else (so for instance the attribute descriptions could live somewhere outside /content, and the files in /content could point to them). I think this would be unfortunate for authors - it's a nice feature of the current organization that all the supplementary files are kept together with the main file.

* don't have supplementary files. Keep everything for a given element in one file, including live samples (not interactive examples of course). This would make these files much bigger and would give us more complicated rules for slicing them up, and would make it harder to reuse content in some ways (since 2 HTML elements, say, couldn't share an example). But it would make other things simpler, and we'd have much fewer files that way of course.

* don't scan for what to build by just walking the tree looking for things. Instead, use something like the "related_content" to tell you what to build. So the build-json starts by thinking "I want to build the HTML docs", so it finds https://github.com/mdn/stumptown-content/blob/master/content/related_content/html.yaml, opens it, resolves it, and walks through the tree building everything in there. I think I like this because it's very explicit, not relying on any kind of magic, and it also guarantees that the tree of built pages, the sidebar, and any index pages are always in sync. Note also that even in the very limited html.yaml file already in there, there are docs (the learning area docs) that aren't under the content/html bit of the tree, but that are needed if you want a complete set of MDN's HTML docs.

**Comment by ddbeck** [comment]
Yeah, there's enough complexity in the file naming thing to not do it in this PR (I kinda thought `<dirname>/<dirname>.md` wouldn't be that big a deal, but the more I think on it, the more issues there are). Let's move on without that for now.

**Comment by wbamberg** [comment]
> Yeah, there's enough complexity in the file naming thing to not do it in this PR (I kinda thought `<dirname>/<dirname>.md` wouldn't be that big a deal, but the more I think on it, the more issues there are). Let's move on without that for now.

OK, sounds good. But I think we will have to face this monster at some point!

**Comment by ddbeck** [comment]
Opened #93 to address the file name question.

**Pull request merged**